dds/DCPS/NetworkResource.cpp

```diff
@@ -90,55 +90,59 @@ String get_fully_qualified_hostname(ACE_INET_Addr* addr)
 
     if (result != 0 || addr_count < 1) {
       ACE_ERROR((LM_ERROR,
-                 ACE_TEXT("(%P|%t) ERROR: Unable to probe network. %p\n"),
+                 ACE_TEXT("(%P|%t) ERROR: get_fully_qualified_hostname: Unable to probe network. %p\n"),
                  ACE_TEXT("ACE::get_ip_interfaces")));
 
     } else {
       for (size_t i = 0; i < addr_count; i++) {
-        VDBG_LVL((LM_DEBUG, "(%P|%t) NetworkResource: found IP interface %C\n", LogAddr::ip(addr_array[i]).c_str()), 4);
+        VDBG_LVL((LM_DEBUG, "(%P|%t) get_fully_qualified_hostname: Found IP interface %C\n", LogAddr::ip(addr_array[i]).c_str()), 4);
       }
 
 #ifdef ACE_HAS_IPV6
-        // Front load IPV6 addresses to give preference to IPV6 interfaces
-        size_t index_last_non_ipv6 = 0;
-        for (size_t i = 0; i < addr_count; i++) {
-          if (addr_array[i].get_type() == AF_INET6) {
-            if (i == index_last_non_ipv6) {
-              ++index_last_non_ipv6;
-            } else {
-              std::swap(addr_array[i], addr_array[index_last_non_ipv6]);
-              ++index_last_non_ipv6;
-            }
+      // Front load IPV6 addresses to give preference to IPV6 interfaces
+      size_t index_last_non_ipv6 = 0;
+      for (size_t i = 0; i < addr_count; i++) {
+        if (addr_array[i].get_type() == AF_INET6) {
+          if (i == index_last_non_ipv6) {
+            ++index_last_non_ipv6;
+          } else {
+            std::swap(addr_array[i], addr_array[index_last_non_ipv6]);
+            ++index_last_non_ipv6;
           }
         }
+      }
 #endif
       for (size_t i = 0; i < addr_count; i++) {
         char hostname[MAXHOSTNAMELEN+1] = "";
 
         // Discover the fully qualified hostname
         if (ACE::get_fqdn(addr_array[i], hostname, MAXHOSTNAMELEN+1) == 0) {
-          VDBG_LVL((LM_DEBUG, "(%P|%t) considering fqdn %C\n", hostname), 4);
-          if (!addr_array[i].is_loopback() && ACE_OS::strchr(hostname, '.') != 0 && choose_single_coherent_address(hostname, false, false) != ACE_INET_Addr()) {
-            VDBG_LVL((LM_DEBUG, "(%P|%t) found fqdn %C from %C\n", hostname, LogAddr(addr_array[i]).c_str()), 2);
-            selected_address = addr_array[i];
-            fullname = hostname;
-            if (addr) {
-              *addr = selected_address;
+          VDBG_LVL((LM_DEBUG, "(%P|%t) get_fully_qualified_hostname: Considering fqdn %C\n", hostname), 4);
+          // Find the first FQDN that resolves to a valid IP interface address.
+          // If there is no such FQDN, consider the list of non-FQDNs.
+          if (ACE_OS::strchr(hostname, '.') != 0) {
+            if (!addr_array[i].is_loopback() &&
```

Comment:
What happens to the candidate name if it passes the first check (if `strchr()` returns non-zero) but fails the second (it's a loopback or choose_single_coherent_address returns `ACE_INET_Addr()`)? Before, we did some checks to see if we needed to add it to non-FQDN, but now it looks like we ignore it completely?